networkx/tests/test_all_random_functions.py

```diff
@@ -18,21 +16,13 @@
 
 def t(f, *args, **kwds):
     """call one function and check if global RNG changed"""
-    global progress
```

Comment:
IIRC this test calls a very large number of tests and there is really no good way to tell which one was the one that failed.  That means there must be a better way to write the test. Some sort of parametrization I suspect. The `global` and prints are quite ugly. But we need to be able to figure out which one of them fails... and these comments help to figure that out.

So, this is another one where it's not clear to me that removing the cruft is worth the cost of remembering how to figure out how to debug it.

There are other improvements that would be helpful here -- like the test has to be manually updated each time a function that uses a random number generator is added to NetworkX -- and we often don't remember to add the function. But that's a separate issue for sure. :}

Reply:
Agreed, these random tests are special in how they're structured. I will go ahead and back these changes out of this PR to see if we can't address this particularly set of tests separately!